src/rebar3.erl

```diff
@@ -320,7 +322,6 @@ ensure_running(App, Caller) ->
     end.
 
 state_from_global_config(Config, GlobalConfigFile) ->
-    rebar_utils:set_httpc_options(),
```

Comment:
Well, I tried to move local `rebar.config` into `~/.config/rebar3/` and haven't found any issues about during tests run. Also `set_httpc_options` seems to be somewhat self sufficient in this respect as the function it calls will re-read global_config: 
```

get_http_vars(Scheme) ->
    OS = case os:getenv(atom_to_list(Scheme)) of
        Str when is_list(Str) -> Str;
        _ -> []
    end,
    GlobalConfigFile = rebar_dir:global_config(),
    Config = rebar_config:consult_file(GlobalConfigFile),
    proplists:get_value(Scheme, Config, OS).

```